File: CS/MainWindowViewModel.cs
using System.Collections.ObjectModel;
using System.Linq;

namespace SplashScreenStartupTest {
    public class MainWindowViewModel {
        public ObservableCollection<Item> Items { get; private set; }
        protected MainWindowViewModel() {
            Items = new ObservableCollection<Item>();
            Enumerable.Range(0, 100)
                .Select(x => new Item() { Id = x, Value = x.ToString() })
                .ToList()
                .ForEach(x => Items.Add(x));
        }
    }
    public class Item {
        public int Id { get; set; }
        public string Value { get; set; }
    }
}

File: VB/MainWindowViewModel.vb
Imports System.Collections.ObjectModel
Imports System.Linq

Namespace SplashScreenStartupTest

    Public Class MainWindowViewModel

        Private _Items As ObservableCollection(Of SplashScreenStartupTest.Item)

        Public Property Items As ObservableCollection(Of Item)
            Get
                Return _Items
            End Get

            Private Set(ByVal value As ObservableCollection(Of Item))
                _Items = value
            End Set
        End Property

        Protected Sub New()
            Items = New ObservableCollection(Of Item)()
            Enumerable.ToList(Of Item)(Enumerable.Select(Of Integer, Global.SplashScreenStartupTest.Item)(Enumerable.Range(CInt(0), CInt(100)), CType(Function(x) CType(New Item() With {.Id = x, .Value = x.ToString()}, Item), System.Func(Of Integer, Item)))).ForEach(Sub(x) Items.Add(x))
        End Sub
    End Class

    Public Class Item

        Public Property Id As Integer

        Public Property Value As String
    End Class
End Namespace
